src/main/java/com/mappy/fpm/batches/tomtom/dbf/timedomains/TimeDomainsParser.java

```diff
@@ -66,6 +66,26 @@ private String parse(TimeDomains timeDomains) {
         throw new IllegalArgumentException("Unable to parse '" + domain + "'");
     }
 
+    private String padLeftZeros(String inputString, int length) {
+        return String.format("%1$" + length + "s", inputString).replace(' ', '0');
+    }
+
+    public String[] parseDateInterval(TimeDomains timeDomain) {
+        String[] dates = new String[2];
+        String pattern = "\\[\\(y(\\d{4})M(\\d{1,2})d(\\d{1,2})\\)\\(y(\\d{4})M(\\d{1,2})d(\\d{1,2})\\)\\]";
```

Comment:
According to the test below, the expected format seems to be `[(y2020M1d15)(y2022M12d5)]`.

It would be cleaner to deduplicate the date parsing.